backend/src/main/java/controllers/MySQLController.java

```diff
@@ -398,20 +398,27 @@ public boolean addNewUser(User newUser) {
                 String userId = likesResult.getString("user_id");
                 // now I just need the post ID
                 String postId = likesResult.getString("post_id");
-                postLike.put(postId, new User(username, password, bio, userId));
+                if (!postLike.containsKey(postId)) {
+                    ArrayList<User> userArray = new ArrayList<User>();
```

Comment:
I'm a little confused what's happening here, would a more descriptive variable name for userArray help maybe?  And same with "postLike", it sounds like that is one like, but it's a hash map. Can you explain what is happening and if you think it is good idea change the variable names?

Reply:
I think it should be "postLikes"?